google-cloud-tools-plugin/ultimate/src/com/google/cloud/tools/intellij/appengine/server/instance/AppEngineServerModel.java

```diff
@@ -124,107 +138,506 @@ public int getDefaultPort() {
+  @Override
+  public String getHost() {
+    return null;
   }
 
-  public void setArtifact(@Nullable Artifact artifact) {
-    if (artifact != null) {
-      myArtifactPointer = ArtifactPointerManager.getInstance(myCommonModel.getProject())
-          .createPointer(artifact);
-    } else {
-      myArtifactPointer = null;
-    }
+  public void setHost(String host) {
+    settings.setHost(host);
   }
 
-  public static class AppEngineModelSettings {
+  @Override
+  public Integer getPort() {
+    return settings.getPort();
+  }
+
+  public void setPort(Integer port) {
+    settings.setPort(port);
+  }
+
+  @Override
+  public String getAdminHost() {
+    return settings.getAdminHost();
+  }
+
+  public void setAdminHost(String adminHost) {
+    settings.setAdminHost(adminHost);
+  }
+
+  @Override
+  public Integer getAdminPort() {
+    return settings.getAdminPort();
+  }
+
+  public void setAdminPort(Integer adminPort) {
+    settings.setAdminPort(adminPort);
+  }
+
+  @Override
+  public String getAuthDomain() {
+    return settings.getAuthDomain();
+  }
+
+  public void setAuthDomain(String authDomain) {
+    settings.setAuthDomain(authDomain);
+  }
+
+  @Override
+  public String getStoragePath() {
+    return settings.getStoragePath();
+  }
+
+  public void setStoragePath(String storagePath) {
+    settings.setStoragePath(storagePath);
+  }
+
+  @Override
+  public String getLogLevel() {
+    return settings.getLogLevel();
+  }
+
+  public void setLogLevel(String logLevel) {
+    settings.setLogLevel(logLevel);
+  }
+
+  @Override
+  public Integer getMaxModuleInstances() {
+    return settings.getMaxModuleInstances();
+  }
+
+  public void setMaxModuleInstances(Integer maxModuleInstances) {
+    settings.setMaxModuleInstances(maxModuleInstances);
+  }
+
+  @Override
+  public Boolean getUseMtimeFileWatcher() {
+    return settings.isUseMtimeFileWatcher();
+  }
+
+  public void setUseMtimeFileWatcher(Boolean useMtimeFileWatcher) {
+    settings.setUseMtimeFileWatcher(useMtimeFileWatcher);
+  }
+
+  @Override
+  public String getThreadsafeOverride() {
+    return settings.getThreadsafeOverride();
+  }
+
+  public void setThreadsafeOverride(String threadsafeOverride) {
+    settings.setThreadsafeOverride(threadsafeOverride);
+  }
+
+  @Override
+  public String getPythonStartupScript() {
+    return settings.getPythonStartupScript();
+  }
+
+  public void setPythonStartupScript(String pythonStartupScript) {
+    settings.setPythonStartupScript(pythonStartupScript);
+  }
+
+  @Override
+  public String getPythonStartupArgs() {
+    return settings.getPythonStartupArgs();
+  }
+
+  public void setPythonStartupArgs(String pythonStartupArgs) {
+    settings.setPythonStartupArgs(pythonStartupArgs);
+  }
+
+  private static final String DELIMITER = "!@#";
+
+  @Override
+  public List<String> getJvmFlags() {
+    return Arrays.asList(settings.getJvmFlags() != null
+        ? settings.getJvmFlags().split(DELIMITER) : new String[]{});
```

Comment:
I wanted to protect myself against the fact that space might be a valid character in a jvm flag but, actually, I don't think it is and I think that's how the flags would be separated. There is even a bug where if the user adds more than one flag separated by spaces, we only count it as one because of how setJvmFlags works.

So, I'm just going to use space as a separator.